src/binary.js

```diff
@@ -119,15 +119,37 @@ function getType(value) {
     }
 }
 
+const minChinkSize = 8;
+
 class Writer {
-    constructor(initialSize) {
-        this.bytes = new Uint8Array(initialSize || 100000000);
-        this.view = new DataView(this.bytes.buffer);
-        this.pos = 0;
+    constructor(chunkSize = 64000) {
+        this.chunks = [];
         this.stringEncoder = new TextEncoder();
+        this.chunkSize = chunkSize < minChinkSize ? minChinkSize : chunkSize;
+        this.createChunk();
     }
     get value() {
-        return this.bytes.subarray(0, this.pos);
+        this.flushChunk();
+        const resultBuff = new Uint8Array(Buffer.concat(this.chunks));
```

Comment:
This will create a copy of `Buffer`'s buffer. I'm doubt we need a transformation here, since `Buffer` inherits from `Uint8Array`. But if we want it, it should be done this way:
```js
const resultBuffer = Buffer.concat(this.chunks);
const result = new Uint8Array(resultBuffer.buffer, resultBuffer.offset, resultBuffer.byteLength);
// ...
return result;
```